VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsThreadFragment.java

```diff
@@ -171,6 +172,24 @@ public void onDestroy() {
         EventBus.getDefault().unregister(this);
     }
 
+    @Override
+    public void onRestart() {
+        if (postsSort == DiscussionPostsSort.LAST_ACTIVITY_AT && mSelectedItem != -1) {
+            // Move the last viewed thread to the top
+            discussionPostsAdapter.moveToTop(mSelectedItem);
+            mSelectedItem = -1;
```

Comment:
The post is immediately moved to the top before the activity transition, even if we post it on the handler. So I guess, current approach is correct.